### PR TITLE
Support casting a `FixedSizedList<T>[1]` to `T`

### PR DESCRIPTION
**Pull request by sadboy**
# Which issue does this PR close?

N/A
This is a followup to #5340 

# Rationale for this change
 

This implements the mirror of #5340 -- casting `FixedSizedList<T>[1]` to `T`. Taken together, this allows `FixedSizedList<T>[1]` to be used "as though it were a `T`" for the purpose of casting.

# What changes are included in this PR?

* New casting method: `cast_single_element_fixed_size_list_to_values`
* Minor bug fix in `FixedSizeListArray::try_new`: allow `nulls` to be `None` when `values` has null buffer with null_count of 0.

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-array/src/array/fixed_size_list_array.rs [hunk 1]
```diff
@@ -183,7 +183,8 @@ impl FixedSizeListArray {
                 || nulls
                     .as_ref()
                     .map(|n| n.expand(size as _).contains(&a))
-                    .unwrap_or_default();
+                    .unwrap_or_default()
+                || (nulls.is_none() && a.null_count() == 0);
```
Comment (tustvold) [inline arrow-array/src/array/fixed_size_list_array.rs:187]: Why this change?

File: arrow-cast/src/cast/mod.rs [hunk 2]
```diff
@@ -153,6 +153,8 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
         (_, LargeList(list_to)) => can_cast_types(from_type, list_to.data_type()),
         (_, FixedSizeList(list_to,size)) if *size == 1 => {
             can_cast_types(from_type, list_to.data_type())},
+        (FixedSizeList(list_from,size), _) if *size == 1 => {
```
Comment (tustvold) [inline arrow-cast/src/cast/mod.rs:156]: I think this could recurse infinitely if you tried to cast a FixedSizeList of type A to a FixedSizeList of type B

File: arrow-array/src/array/fixed_size_list_array.rs (hunk 1 above)
Comment (sadboy) [reply arrow-array/src/array/fixed_size_list_array.rs:187]: The `values` array could have a `NullBuffer` but no actual `null`s in it (which happens when you safe-cast from a primitive array). This just accounts for that case.
